From 9f9609ea9ec1b22a739e56cb4824a4765ace7422 Mon Sep 17 00:00:00 2001
From: "Yi, Hongyu" <hongyu.yi@intel.com>
Date: Fri, 25 May 2018 10:20:30 +0800
Subject: [PATCH 062/129] media: intel-ipu4: isys: Refining SOF/EOF trace event
 for CSI

Message for Open Source:
SOF/EOF trace event is fixed/added for CSI.

Message for Internal:
[Issue/Feature]
SOF/EOF trace event is wrong/missed for CSI.
[Root Cause/Changes]
SOF/EOF trace event is fixed/added for CSI.

Change-Id: I1f9d2dbc086101ecc18e2ae6cd6314835e6eddd6
Tracked-On: #JKRLCSN-268
Signed-off-by: Yi, Hongyu <hongyu.yi@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-isys-csi2.c   |  3 +++
 drivers/media/pci/intel/ipu-trace-event.h | 17 +++++++++++++++++
 2 files changed, 20 insertions(+)

diff --git a/drivers/media/pci/intel/ipu-isys-csi2.c b/drivers/media/pci/intel/ipu-isys-csi2.c
index e6d5707..c5fc878 100644
--- a/drivers/media/pci/intel/ipu-isys-csi2.c
+++ b/drivers/media/pci/intel/ipu-isys-csi2.c
@@ -19,6 +19,7 @@
 
 #define CREATE_TRACE_POINTS
 #define IPU_SOF_SEQID_TRACE
+#define IPU_EOF_SEQID_TRACE
 #include "ipu-trace-event.h"
 
 static const u32 csi2_supported_codes_pad_sink[] = {
@@ -901,6 +902,8 @@ void ipu_isys_csi2_eof_event(struct ipu_isys_csi2 *csi2, unsigned int vc)
 	if (ip) {
 		frame_sequence = atomic_read(&ip->sequence);
 
+		trace_ipu_eof_seqid(frame_sequence, csi2->index, vc);
+
 		dev_dbg(&csi2->isys->adev->dev,
 			"eof_event::csi2-%i sequence: %i, vc: %d, stream_id: %d\n",
 			csi2->index, frame_sequence, vc, ip->stream_id);
diff --git a/drivers/media/pci/intel/ipu-trace-event.h b/drivers/media/pci/intel/ipu-trace-event.h
index b71ca24..173c931 100644
--- a/drivers/media/pci/intel/ipu-trace-event.h
+++ b/drivers/media/pci/intel/ipu-trace-event.h
@@ -26,6 +26,23 @@
 	);
 #endif
 
+#ifdef IPU_EOF_SEQID_TRACE
+TRACE_EVENT(ipu_eof_seqid,
+	    TP_PROTO(unsigned int seqid, unsigned int csiport,
+		     unsigned int csivc),
+	    TP_ARGS(seqid, csiport, csivc),
+	    TP_STRUCT__entry(__field(unsigned int, seqid)
+			     __field(unsigned int, csiport)
+			     __field(unsigned int, csivc)
+	    ),
+	    TP_fast_assign(__entry->seqid = seqid;
+			   __entry->csiport = csiport;
+			   __entry->csivc = csivc;),
+	    TP_printk("seqid<%u>,csiport<%u>,csivc<%u>", __entry->seqid,
+		      __entry->csiport, __entry->csivc)
+	);
+#endif
+
 #ifdef IPU_PERF_REG_TRACE
 TRACE_EVENT(ipu_perf_reg,
 	    TP_PROTO(unsigned int addr, unsigned int val),
-- 
1.9.1

